MathML <mroot>: clamp RadicalKernBefore/AfterDegree

fred-wang opened the pull request:
#### 07c9448d9a71ea3d6fe2c858bb96ae66fb54a664
<pre>
MathML &lt;mroot&gt;: clamp RadicalKernBefore/AfterDegree
<a href="https://bugs.webkit.org/show_bug.cgi?id=276427">https://bugs.webkit.org/show_bug.cgi?id=276427</a>

Reviewed by Rob Buis.

Implements the clamping described in MathML Core.
See <a href="https://w3c.github.io/mathml-core/#root-with-index">https://w3c.github.io/mathml-core/#root-with-index</a>

* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/radic3als/root-parameters-2-expected.txt: This test passes.
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/not-participating-to-parent-layout-expected.txt: tweak the failure message.
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/width-height-001-expected.txt: ditto.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/width-height-001-expected.txt: ditto.
* LayoutTests/platform/glib/mathml/presentation/roots-expected.txt: update render tree expectation.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/mathml/relations/css-styling/not-participating-to-parent-layout-expected.txt: tweak the failure message.
* LayoutTests/platform/ios/mathml/presentation/roots-expected.txt: update render tree expectation.
* LayoutTests/platform/mac/mathml/presentation/roots-expected.txt: update render tree expectation.
* Source/WebCore/rendering/mathml/RenderMathMLRoot.cpp:
(WebCore::RenderMathMLRoot::horizontalParameters): add a indexWidth parameter and perform the clamping fromt the spec.
(WebCore::RenderMathMLRoot::computePreferredLogicalWidths): pass the preferred width of the index to the helper.
(WebCore::RenderMathMLRoot::layoutBlock): pass the width of the index to the helper.
(WebCore::RenderMathMLRoot::paint): ditto.
* Source/WebCore/rendering/mathml/RenderMathMLRoot.h: add indexWidth parameter too.

Canonical link: <a href="https://commits.webkit.org/280902@main">https://commits.webkit.org/280902@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/775bbb5011d4812ddfcb94a8e047783179a1ef31

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/57989 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/37317 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/10465 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/61614 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/8434 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/60117 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/44953 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/8622 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/46994 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/6007 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/60019 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/34997 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/50124 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/27825 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/31761 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/7419 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/7438 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/53707 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/7687 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/63302 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/1903 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/7756 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/54218 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/1910 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/50135 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/54357 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12825 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/1630 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/33146 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/34232 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/35316 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/33977 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rwlbuis reviewed (approved):
LGTM. Note, seems like commit message lacks the Changelog part regarding the files changed.

fred-wang committed:
MathML <mroot>: clamp RadicalKernBefore/AfterDegree
https://bugs.webkit.org/show_bug.cgi?id=276427

Reviewed by Rob Buis.

Implements the clamping described in MathML Core.
See https://w3c.github.io/mathml-core/#root-with-index

* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/radic3als/root-parameters-2-expected.txt: This test passes.
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/not-participating-to-parent-layout-expected.txt: tweak the failure message.
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/width-height-001-expected.txt: ditto.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/width-height-001-expected.txt: ditto.
* LayoutTests/platform/glib/mathml/presentation/roots-expected.txt: update render tree expectation.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/mathml/relations/css-styling/not-participating-to-parent-layout-expected.txt: tweak the failure message.
* LayoutTests/platform/ios/mathml/presentation/roots-expected.txt: update render tree expectation.
* LayoutTests/platform/mac/mathml/presentation/roots-expected.txt: update render tree expectation.
* Source/WebCore/rendering/mathml/RenderMathMLRoot.cpp:
(WebCore::RenderMathMLRoot::horizontalParameters): add a indexWidth parameter and perform the clamping fromt the spec.
(WebCore::RenderMathMLRoot::computePreferredLogicalWidths): pass the preferred width of the index to the helper.
(WebCore::RenderMathMLRoot::layoutBlock): pass the width of the index to the helper.
(WebCore::RenderMathMLRoot::paint): ditto.
* Source/WebCore/rendering/mathml/RenderMathMLRoot.h: add indexWidth parameter too.

Canonical link: https://commits.webkit.org/280902@main
